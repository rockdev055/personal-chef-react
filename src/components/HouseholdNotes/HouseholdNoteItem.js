import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';
import { List } from 'semantic-ui-react';

export default function HouseholdNoteItem({ note }) {
  if (!note) return null;
  return (
    <List.Item data-testid="household-note-item" key={note.id}>
      {format(new Date(note.created_at), 'MMMM Do, YYYY')} - {note.content}
    </List.Item>
  );
}

HouseholdNoteItem.propTypes = {
  note: PropTypes.shape({
    created_at: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    content: PropTypes.string.isRequired,
  }),
};
