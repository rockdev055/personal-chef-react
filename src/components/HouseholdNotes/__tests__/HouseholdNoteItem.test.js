import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HouseholdNoteItem from '../HouseholdNoteItem';

afterEach(cleanup);

describe('<HouseholdNoteItem /> component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<HouseholdNoteItem />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('does not render if a note prop is not provided', () => {
    const { queryByTestId } = render(<HouseholdNoteItem />);
    expect(queryByTestId('household-note-item')).toBeFalsy();
  });

  it('renders if a note prop is provided', () => {
    const note = { id: 2, created_at: '2018-11-05T22:37:48.340Z', content: 'Spoke with the client again.' };
    const { queryByTestId } = render(<HouseholdNoteItem note={note} />);
    expect(queryByTestId('household-note-item')).toBeTruthy();
  });
});
