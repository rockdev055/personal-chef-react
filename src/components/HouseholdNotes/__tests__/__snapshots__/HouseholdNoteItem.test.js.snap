// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HouseholdNoteItem /> component matches the snapshot 1`] = `<!-- react-empty: 1 -->`;
